teos/src/watcher.rs

```diff
@@ -439,12 +478,13 @@ impl Watcher {
     ///
     /// If the appointment is rejected by the [Responder] (i.e. for being invalid), the data is wiped
     /// from the database but the slot is not freed.
-    fn store_triggered_appointment(
-        &self,
```

Comment:
Yeah this didn't feel right to me either but based on my understanding of Rust this is inevitable because you cannot use `self` within an `async` call as it violates the `static` lifetime requirement.